core/arch/arm/include/kernel/spmc_sp_handler.h

```diff
@@ -17,6 +17,23 @@
 void spmc_sp_thread_entry(uint32_t a0, uint32_t a1, uint32_t a2, uint32_t a3);
 void spmc_sp_msg_handler(struct thread_smc_args *args,
 			 struct sp_session *caller_sp);
+TEE_Result spmc_sp_add_share(struct ffa_mem_transaction *input_descr,
+			     uint64_t global_handle, size_t blen);
+
+struct sp_shared_mem {
+	size_t counter;
+	struct shared_mem *s_mem;
+	struct ffa_mem_access *access_descr;
```

Comment:
Overall I'd rather avoid using the structs which work directly with the FF-A ABI unless they are a perfect match. In this case it doesn't seem like a perfect match